### ISSUE TITLE
version 26 breaks the api

### ISSUE THREAD

**Issue by csicky** [ag-grid/ag-grid#4647]
Until now you used to put warnings in console if I used something which was deprecated. You added the warning about `addRowCompListener`, which I actually use to be able to know when a row is not there anymore to remove an event handler from it like this: 

```
this.params.node.addEventListener('mouseEnter', this.setUpInfoComponent)

this.params.addRowCompListener('virtualRowRemoved', () => {
  this.params.node.removeEventListener('mouseEnter', this.setUpInfoComponent)
})
```
I can't say for sure this is the best way to do the above, but it worked until now. However this is not too much a problem compared to the following.

The inner api seems to be completely changed and no warnings in console tells me what happened. We use a paid version of ag grid, we use a lot of custom components and we rely heavily on the gridApi. I wish there were warnings telling me what changed to not have to hunt around.

![image](https://user-images.githubusercontent.com/3931907/131538948-5b74e804-fdc6-45b5-b16e-d6a5f6d48b41.png)
![image](https://user-images.githubusercontent.com/3931907/131538846-5eaa2ae3-9612-4fa5-b62c-297cb0d9b724.png)
![image](https://user-images.githubusercontent.com/3931907/131542511-b7610dac-5812-45f4-8068-c3a735a54a62.png)

This is not even announced in the change log under breaking changes. This is the worst upgrade of ag grid ever. When we upgraded to 23, yes we had to do some work, but it was all nice and documented, now it is not nice at all.

We use ag grid since many years and until now I learned a lot, I spent so much time in the documentation, but now that too seems to be changed, so now I have to start over all the learning because I can't tell what are the differences. 

**TLDR:
Please add more warnings in the console about grid api changes.**

**Comment by csicky** [comment]
I think we will stay at version 25 for a while, I should have learned until now that version x.0.0 is in fact beta.

**Comment by csicky** [comment]
Ok, so the only thing that happened is the `gridApi.columnController` was renamed to `gridApi.columnModel` and that broke a lot of things. Now after a search replace everything seems to be working again.
